### PR TITLE
c64_cass.xml: 13 new dumps

### PR DESCRIPTION
**Pull request by scoomby**
New working software list additions
---------------------------------------
Bigfoot (Codemasters) [C64 Ultimate Tape Archive V2.0]
Bigtop Barney (Interceptor Software) [C64 Ultimate Tape Archive V2.0]
Black Lamp (Firebird) [C64 Ultimate Tape Archive V2.0]
Black Tiger (U.S. Gold) [C64 Ultimate Tape Archive V2.0]
Blood Brothers (Prism Leisure) [C64 Ultimate Tape Archive V2.0]
Blue Max (U.S. Gold) [C64 Ultimate Tape Archive V2.0]
Blue Max 2001 (U.S. Gold) [C64 Ultimate Tape Archive V2.0]
The Blues Brothers (Titus) [C64 Ultimate Tape Archive V2.0]
BMX Simulator 2 (Codemasters) [C64 Ultimate Tape Archive V2.0]
Bobby Bearing (The Edge) [C64 Ultimate Tape Archive V2.0]
The Boggit: Bored Too (CRL) [C64 Ultimate Tape Archive V2.0]
Bomb Jack (Elite Systems) [C64 Ultimate Tape Archive V2.0]
Bomb Jack II (Elite Systems) [C64 Ultimate Tape Archive V2.0]

NB. I noticed that The Boggit will not always load with the UI enabled (it sometimes gets stuck displaying "FOUND BOGGIT PART 1").  With UI disabled straight after initiating "Play" in Tape Control, the game always loads correctly.  I will investigate this further.

### CODE REVIEW

**Comment by Pernod70** [comment]
Why did you start changing all the filenames to match the shortname? The original filenames are far more descriptive of what they actually are.

**Comment by scoomby** [comment]
> Why did you start changing all the filenames to match the shortname? The original filenames are far more descriptive of what they actually are.

This was due to comments that were given on my earlier PR's (e.g. https://github.com/mamedev/mame/pull/8331)

**Comment by scoomby** [comment]
I also put a proposal to improve the software list submission standards ref. https://github.com/mamedev/mame/pull/8357

**Comment by Pernod70** [comment]
I don't see anything in those previous comments that says the filename should match the shortname. The original filenames match the cassette labels, except where we have alternative versions, and should be kept in my opinion.

**Comment by scoomby** [comment]
> I don't see anything in those previous comments that says the filename should match the shortname. The original filenames match the cassette labels, except where we have alternative versions, and should be kept in my opinion.

Agreed, but I was asked to greatly improve the filenames so I put a proposal together and the last two PR's with the shortened filenames have been accepted.  The shortened filename approach can also be found in other software list roms (e.g. 5200, 7800, coleco, vectrex, etc.) so it's not unique.  I agree with Vas Crabb, the software list roms are in a mess and there is no consistency.  For example, some software list rom filenames include the release year which isn't required as we have metadata fields for this.  Before the shortened filename approach, my previous PR's which basically followed the same standard as other C64_cass software list submissions but I was advised that this was no longer acceptable.  Note that I would have been happy to carry on using the filenames as per the C64 Ultimate Tape Archive (where possible) as it takes now taking more time to do these PR's.

**Comment by DavidHaywood** [comment]
there's no reason to be using less descriptive filenames

**Comment by scoomby** [comment]
> there's no reason to be using less descriptive filenames
> 
> for files sourced elsewhere my personal preference is to use their naming scheme, it makes it easier to cross-reference.

Hi David.  Thanks for your input.  But like I said, that is what I tried to do initially (it was my preference too) but that was considered to be no longer acceptable and the software list rom standards needed to improve greatly which is what I have tried to do.  At the moment, whatever I submit isn't acceptable right now!!

**Comment by Tafoid** [comment]
Keeping them as close to the originally source filenames or chip labels just makes sense for attempting to be as accurate as possible.  To purposely dumb down the actual filenames makes little sense at this point.  When MAME started, that is all there was available was limited by the OS (8.3 filenames) and we've over time been pushing that away.  Short names for launching machines are a different matter and while you can use 16 characters now (even for parents), it is still advised to keep them as short as possible for ease of development.

In short, keep filename what you find them as with the exceptions of multiple different images using the same filename (which happens rarely).  Adding a tag or something unique that can identify that unique set isn't unheard of.

**Comment by scoomby** [comment]
> Keeping them as close to the originally source filenames or chip labels just makes sense for attempting to be as accurate as possible. To purposely dumb down the actual filenames makes little sense at this point. When MAME started, that is all there was available was limited by the OS (8.3 filenames) and we've over time been pushing that away. Short names for launching machines are a different matter and while you can use 16 characters now (even for parents), it is still advised to keep them as short as possible for ease of development.
> 
> In short, keep filename what you find them as with the exceptions of multiple different images using the same filename (which happens rarely). Adding a tag or something unique that can identify that unique set isn't unheard of.

Scott, that is exactly what I was doing but I was told it was no longer acceptable (refer to PR https://github.com/mamedev/mame/pull/8331).  I know the initial discussion was relating to the use of "(alt)" with clones that have the same filenames. It was just to get around the issue when clones are merged with the original sets (you can't have exactly the same filenames).  However, the the conversation grew to something more than that and it was about the software list standards in general.  I just need a clear "goalpost" with respect the software list filename standards that doesn't move and I will keep to it!!

**Review by cuavas — CHANGES_REQUESTED** [review]
You seem to have ignored or badly misinterpreted what I said.  I was objecting to two things:
* Adding metadata into the labels for the tapes (the `name` attributes of the `rom` elements) – we have a schema that allows adding that elsewhere
* Adding disambiguation text to the labels for clone sets where the clone set tape has the same label as the parent

The labels (`name` attribute values on `rom` elements) should match what the actual labels on the physical tapes say, within the limit of the character set allowed.

**Comment by scoomby** [comment]
> You seem to have ignored or badly misinterpreted what I said. I was objecting to two things:
> 
> * Adding metadata into the labels for the tapes (the `name` attributes of the `rom` elements) – we have a schema that allows adding that elsewhere
> * Adding disambiguation text to the labels for clone sets where the clone set tape has the same label as the parent
> 
> The labels (`name` attribute values on `rom` elements) should match what the actual labels on the physical tapes say, within the limit of the character set allowed.

I probably have misinterpreted your points but your points above are much clearer.  Previously, the only time I have added something to the filename like "(alt)" or "(kixx)" is for a clone set where the filenames from the C64 Ultimate Tape Archive are exactly the same for the original and the clone sets.  If this the only issue, please provide direction on how to work with merged sets where filenames are identical for the original and clone sets.  Thanks in advance.

**Comment by scoomby** [comment]
Here is an example of 2 sets with the same filename:
Ultimate_Tape_Archive_V2.0\Breakthru_(1986_ERBE_Software)_[6162]\Breakthru.tap
Ultimate_Tape_Archive_V2.0\Breakthru_(1987_U.S._Gold)_[750]\Breakthru.tap

**Comment by cuavas** [comment]
We’ve been through this before, but here goes…
* MAME searches archives for files with the expected CRC – unless there are multiple files in the set with the same CRC but different content, you can call the files whatever you want in an archived set, the file names in archives don’t actually have to match the labels
* When matching on name and CRC in an archive, MAME matches against the last path component – you can have “a/name” and “b/name” inside your archive and MAME will still find them
* There’s actually nothing in the PKzip format that prevents you from having multiple archive members with the same name – you can create an archive containing two files with the same name if you really want to
* Convenience for people using merged sets is not the primary consideration

**Comment by cuavas** [comment]
To make it absolutely clear: **there is no requirement for labels to be unique across parent and clone sets**.  It’s not unusual to have ROMs with the same label but different content across a set of clones in MAME.

**Comment by DavidHaywood** [comment]
> Convenience for people using merged sets is not the primary consideration

This continues to be one of the most universally hated MAME policies.  For tapes it doesn't even make sense as the tape labels will always be different in some ways between tapes.

**Comment by cuavas** [comment]
> > Convenience for people using merged sets is not the primary consideration
> 
> This continues to be one of the most universally hated MAME policies. For tapes it doesn't even make sense as the tape labels will always be different in some ways between tapes.

The label *isn’t* necessarily different between tapes – there are apparently multiple releases of the same software with the same thing printed on the tape labels.

**Comment by scoomby** [comment]
Thanks for clarifying and direction.  Note that in quite a lot of cases, the later release by say a budget brand isn't exactly the same as the original.  Sometimes the loader software, loading picture/music is different.  There are sometimes regional differences too.

**Comment by scoomby** [comment]
> > > Convenience for people using merged sets is not the primary consideration
> > 
> > 
> > This continues to be one of the most universally hated MAME policies. For tapes it doesn't even make sense as the tape labels will always be different in some ways between tapes.
> 
> The label _isn’t_ necessarily different between tapes – there are apparently multiple releases of the same software with the same thing printed on the tape labels.

That is correct.  There are many examples of this hence the tape filenames being the same as the labels are the same.

**Comment by scoomby** [comment]
I have updated the rom filenames to match the filenames used in the C64 Ultimate Archive V2 set.

**Review by Pernod70 — APPROVED** [review]
